File: 21561 - Suvam Tamang/Week 1/94_Binary Tree Inorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    public List<Integer> inorderTraversal(TreeNode root) {
        
        List<Integer> result = new ArrayList<Integer>();
        

        Stack<TreeNode> stack = new Stack<>();
        TreeNode current = root;

        if (root == null){
            return result;
        }
        
        while(current!=null || !stack.empty()){
            if(current==null){
                current = stack.pop();
                result.add(current.val);
                current = current.right;
            }
            else{
                stack.push(current);
                current = current.left;
            }
        }
   
       
        return result;
    }
}

File: 21558 - Nithin/week 1/94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> ans;
    vector<int> inorderTraversal(TreeNode* root) {
        stack<TreeNode*> stck;
        TreeNode* curr_node = root;
        while(stck.empty() == false || curr_node != nullptr){
            if(curr_node != nullptr){
                stck.push(curr_node);
                curr_node = curr_node->left;
            }
            else{
                curr_node = stck.top();
                stck.pop();
                ans.push_back(curr_node->val);
                curr_node = curr_node->right;
            }
        }
        return ans;
    
    }
};